code/script.js

```diff
@@ -222,104 +226,195 @@ const generateBoard = () => {
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+  } else if (category === 'hair') {
+  // Similar to the one above
+    if (keep) {
+    keepCharactersHairEyes (category, value)
+    alert ( `Yes, the person has ${value} hair! Keep all people with ${value} hair`)
+    } else {
+    removeCharactersHairEyes (category, value)
+    alert (`No, the person doesn't have ${value} hair! Remove all people with ${value} hair`)
+    }
+  
+  } else if (category === 'eyes') {
+    // Similar to the one above
+      if (keep) {
+      keepCharactersHairEyes (category, value)
+      alert ( `Yes, the person has ${value} eyes! Keep all people with ${value} eyes`)
+      } else {
+      removeCharactersHairEyes (category, value)
+      alert (`No, the person doesn't have ${value} eyes! Remove all people with ${value} eyes`)
+      }
+    }
+  // This function redraws the board with the remaining people.
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+
+// When clicking guess, the player first have to confirm that they want to make a guess. Store the interaction from the player in a variable.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+
+const userConfirmation = confirm("You sure? Please confirm!")
+if (userConfirmation == true) {
+   // If the player wants to guess, invoke the checkMyGuess function.
+  checkMyGuess(personToConfirm)
+} else {
+  // code here for else-statement?
+} 
 }
 
-// If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  // If user confirms, the checkMyGuess function is invoked
+  
+// Checks if the personToCheck is the same as the secret person's name
+const checkMyGuess = (personToConfirm) => {
+  if(secret.name === personToConfirm) {
+    // Shows the win or lose section
+    winOrLose.style.display = "block"
+    // Hides the game board
+    board.style.display = "none"
+    // Message to show in the win section
+    winOrLoseText.innerHTML =
+    `<h1>Yaaayy you won!!</h1>`
+  } else {
```

Comment:
On my cellphone part of the guess who logo is outside of the picture, i would put some padding above!